### PR TITLE
Fix event streams in the orchestrator implementation

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
This PR gets event streams working in the client orchestrator implementation, and depends on #2671.

The orchestrator's `TypeErasedBox` enforces a `Send + Sync` requirement on inputs and outputs. For the most part, this isn't an issue since almost all generated inputs/outputs are `Send + Sync`, but it turns out the `EventStreamSender` wasn't `Sync` due to an omission of the `Sync` bound. Thus, this PR is a breaking change, as it adds a `Sync` requirement for anyone who passes a stream to an event stream operation.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by jdisanti** [comment]
Requesting server review since this breaking change will impact generated servers as well.

File: aws/rust-runtime/aws-runtime/src/auth/sigv4.rs [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+        } else {
+            SessionTokenMode::Include
+        };
+        settings.signature_location = match operation_config.signing_options.signature_type {
+            HttpSignatureType::HttpRequestHeaders => SignatureLocation::Headers,
+            HttpSignatureType::HttpRequestQueryParams => SignatureLocation::QueryParams,
+        };
+        settings.expires_in = operation_config.signing_options.expires_in;
+        settings
+    }
```
Comment (Velfi) [inline aws/rust-runtime/aws-runtime/src/auth/sigv4.rs:157]: I felt like writing this in a fancy way for some reason:
```suggestion
    fn settings(operation_config: &SigV4OperationSigningConfig) -> SigningSettings {
        let mut settings = SigningSettings::default();
        let SigningOptions {
            double_uri_encode,
            content_sha256_header,
            normalize_uri_path,
            omit_session_token,
            signature_type,
            expires_in,
            ..
        } = operation_config.signing_options;
        settings.percent_encoding_mode = double_uri_encode
            .then_some(PercentEncodingMode::Double)
            .unwrap_or(PercentEncodingMode::Single);
        settings.payload_checksum_kind = content_sha256_header
            .then_some(PayloadChecksumKind::XAmzSha256)
            .unwrap_or(PayloadChecksumKind::NoHeader);
        settings.uri_path_normalization_mode = normalize_uri_path
            .then_some(UriPathNormalizationMode::Enabled)
            .unwrap_or(UriPathNormalizationMode::Disabled);
        settings.session_token_mode = omit_session_token
            .then_some(SessionTokenMode::Exclude)
            .unwrap_or(SessionTokenMode::Include);
        settings.signature_location = match signature_type {
            HttpSignatureType::HttpRequestHeaders => SignatureLocation::Headers,
            HttpSignatureType::HttpRequestQueryParams => SignatureLocation::QueryParams,
        };
        settings.expires_in = expires_in;
        settings
    }
```

**Review by Velfi — APPROVED** [review]

File: aws/rust-runtime/aws-runtime/src/auth/sigv4.rs (hunk 1 above)
Comment (jdisanti) [reply aws/rust-runtime/aws-runtime/src/auth/sigv4.rs:157]: I forgot they stabilized `then_some`. Turns out clippy doesn't like it 😆 
```
warning: use of `.then_some(..).unwrap_or(..)` can be written more clearly with `if .. else ..`
   --> aws-runtime/src/auth/sigv4.rs:145:39
    |
145 |           settings.session_token_mode = operation_config
    |  _______________________________________^
146 | |             .signing_options
147 | |             .omit_session_token
148 | |             .then_some(SessionTokenMode::Exclude)
149 | |             .unwrap_or(SessionTokenMode::Include);
    | |_________________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#obfuscated_if_else
help: try
    |
145 ~         settings.session_token_mode = if operation_config
146 +             .signing_options
147 ~             .omit_session_token { SessionTokenMode::Exclude } else { SessionTokenMode::Include };
    |
```

Comment (Velfi) [reply aws/rust-runtime/aws-runtime/src/auth/sigv4.rs:157]: 🤷‍♀️ Clippy knows best, I suppose.

**Pull request merged**